File: zig/src/rulesets/consoleRenderers.zig
const std = @import("std");
const RendererSet = @import("../types/RendererSet.zig").RendererSet;
const Renderer = @import("../types/Renderer.zig").Renderer;

const alertRenderer = @import("../render-console/consoleAlertRenderer.zig").consoleAlertRenderer;
const blockQuoteRenderer = @import("../render-console/consoleBlockQuoteRenderer.zig").consoleBlockQuoteRenderer;
const codeBlockRenderer = @import("../render-console/consoleCodeBlockRenderer.zig").consoleCodeBlockRenderer;
const codeFenceRenderer = @import("../render-console/consoleCodeFenceRenderer.zig").consoleCodeFenceRenderer;
const codeSpanRenderer = @import("../render-console/consoleCodeSpanRenderer.zig").consoleCodeSpanRenderer;
const commentRenderer = @import("../render-console/consoleCommentRenderer.zig").consoleCommentRenderer;
const deletionRenderer = @import("../render-console/consoleDeletionRenderer.zig").consoleDeletionRenderer;
const emphasisRenderer = @import("../render-console/consoleEmphasisRenderer.zig").consoleEmphasisRenderer;
const footnoteRenderer = @import("../render-console/consoleFootnoteRenderer.zig").consoleFootnoteRenderer;
const footnoteRefRenderer = @import("../render-console/consoleFootnoteRefRenderer.zig").consoleFootnoteRefRenderer;
const footnoteListRenderer = @import("../render-console/consoleFootnoteListRenderer.zig").consoleFootnoteListRenderer;
const hardBreakRenderer = @import("../render-console/consoleHardBreakRenderer.zig").consoleHardBreakRenderer;
const headingRenderer = @import("../render-console/consoleHeadingRenderer.zig").consoleHeadingRenderer;
const headingUnderlineRenderer = @import("../render-console/consoleHeadingUnderlineRenderer.zig").consoleHeadingUnderlineRenderer;
const highlightRenderer = @import("../render-console/consoleHighlightRenderer.zig").consoleHighlightRenderer;
const htmlBlockRenderer = @import("../render-console/consoleHtmlBlockRenderer.zig").consoleHtmlBlockRenderer;
const htmlSpanRenderer = @import("../render-console/consoleHtmlSpanRenderer.zig").consoleHtmlSpanRenderer;
const imageRenderer = @import("../render-console/consoleImageRenderer.zig").consoleImageRenderer;
const insertionRenderer = @import("../render-console/consoleInsertionRenderer.zig").consoleInsertionRenderer;
const linkRenderer = @import("../render-console/consoleLinkRenderer.zig").consoleLinkRenderer;
const listBulletedRenderer = @import("../render-console/consoleListBulletedRenderer.zig").consoleListBulletedRenderer;
const listOrderedRenderer = @import("../render-console/consoleListOrderedRenderer.zig").consoleListOrderedRenderer;
const listTaskItemRenderer = @import("../render-console/consoleListTaskItemRenderer.zig").consoleListTaskItemRenderer;
const paragraphRenderer = @import("../render-console/consoleParagraphRenderer.zig").consoleParagraphRenderer;
const strikethroughRenderer = @import("../render-console/consoleStrikethroughRenderer.zig").consoleStrikethroughRenderer;
const strongRenderer = @import("../render-console/consoleStrongRenderer.zig").consoleStrongRenderer;
const tableRenderer = @import("../render-console/consoleTableRenderer.zig").consoleTableRenderer;
const textRenderer = @import("../render-console/consoleTextRenderer.zig").consoleTextRenderer;
const thematicBreakRenderer = @import("../render-console/consoleThematicBreakRenderer.zig").consoleThematicBreakRenderer;

pub const consoleRenderers = RendererSet{
    .renderers = &.{},
};

pub fn init(allocator: std.mem.Allocator) !RendererSet {
    const renderers = try allocator.alloc(*const Renderer, 30);
    renderers[0] = &alertRenderer;
    renderers[1] = &blockQuoteRenderer;
    renderers[2] = &codeBlockRenderer;
    renderers[3] = &codeFenceRenderer;
    renderers[4] = &codeSpanRenderer;
    renderers[5] = &commentRenderer;
    renderers[6] = &deletionRenderer;
    renderers[7] = &emphasisRenderer;
    renderers[8] = &footnoteRenderer;
    renderers[9] = &footnoteRefRenderer;
    renderers[10] = &footnoteListRenderer;
    renderers[11] = &hardBreakRenderer;
    renderers[12] = &headingRenderer;
    renderers[13] = &headingUnderlineRenderer;
    renderers[14] = &highlightRenderer;
    renderers[15] = &htmlBlockRenderer;
    renderers[16] = &htmlSpanRenderer;
    renderers[17] = &imageRenderer;
    renderers[18] = &insertionRenderer;
    renderers[19] = &linkRenderer;
    renderers[20] = &listBulletedRenderer;
    renderers[21] = &listOrderedRenderer;
    renderers[22] = &listTaskItemRenderer;
    renderers[23] = &paragraphRenderer;
    renderers[24] = &strikethroughRenderer;
    renderers[25] = &strongRenderer;
    renderers[26] = &tableRenderer;
    renderers[27] = &textRenderer;
    renderers[28] = &thematicBreakRenderer;
    renderers[29] = &htmlBlockRenderer;

    return RendererSet{
        .renderers = renderers,
    };
}

pub fn deinit(renderers: *const RendererSet, allocator: std.mem.Allocator) void {
    allocator.free(renderers.renderers);
}

File: web/src/rulesets/consoleRenderers.ts
import alertRenderer from "../render-console/alertRenderer";
import blockQuoteRenderer from "../render-console/blockQuoteRenderer";
import codeBlockRenderer from "../render-console/codeBlockRenderer";
import codeFenceRenderer from "../render-console/codeFenceRenderer";
import codeSpanRenderer from "../render-console/codeSpanRenderer";
import commentRenderer from "../render-console/commentRenderer";
import deletionRenderer from "../render-console/deletionRenderer";
import emphasisRenderer from "../render-console/emphasisRenderer";
import footnoteListRenderer from "../render-console/footnoteListRenderer";
import footnoteRefRenderer from "../render-console/footnoteRefRenderer";
import footnoteRenderer from "../render-console/footnoteRenderer";
import hardBreakRenderer from "../render-console/hardBreakRenderer";
import headingRenderer from "../render-console/headingRenderer";
import headingUnderlineRenderer from "../render-console/headingUnderlineRenderer";
import highlightRenderer from "../render-console/highlightRenderer";
import htmlBlockRenderer from "../render-console/htmlBlockRenderer";
import htmlSpanRenderer from "../render-console/htmlSpanRenderer";
import imageRenderer from "../render-console/imageRenderer";
import insertionRenderer from "../render-console/insertionRenderer";
import linkRenderer from "../render-console/linkRenderer";
import listBulletedRenderer from "../render-console/listBulletedRenderer";
import listOrderedRenderer from "../render-console/listOrderedRenderer";
import listTaskItemRenderer from "../render-console/listTaskItemRenderer";
import paragraphRenderer from "../render-console/paragraphRenderer";
import streamMarkRenderer from "../render-console/streamMarkRenderer";
import strikethroughRenderer from "../render-console/strikethroughRenderer";
import strongRenderer from "../render-console/strongRenderer";
//import subscriptRenderer from "../render-console/subscriptRenderer";
//import superscriptRenderer from "../render-console/superscriptRenderer";
import tableRenderer from "../render-console/tableRenderer";
import textRenderer from "../render-console/textRenderer";
import thematicBreakRenderer from "../render-console/thematicBreakRenderer";
import type Renderer from "../types/Renderer";

const renderers: Renderer[] = [
	alertRenderer,
	blockQuoteRenderer,
	codeBlockRenderer,
	codeFenceRenderer,
	codeSpanRenderer,
	commentRenderer,
	deletionRenderer,
	emphasisRenderer,
	footnoteRenderer,
	footnoteRefRenderer,
	footnoteListRenderer,
	hardBreakRenderer,
	headingRenderer,
	headingUnderlineRenderer,
	highlightRenderer,
	htmlBlockRenderer,
	htmlSpanRenderer,
	imageRenderer,
	insertionRenderer,
	linkRenderer,
	listBulletedRenderer,
	listOrderedRenderer,
	listTaskItemRenderer,
	paragraphRenderer,
	streamMarkRenderer,
	strikethroughRenderer,
	strongRenderer,
	//subscriptRenderer,
	//superscriptRenderer,
	tableRenderer,
	textRenderer,
	thematicBreakRenderer,
];

export default renderers;
